src/test/regress/sql/multi_metadata_sync.sql

```diff
@@ -380,37 +380,121 @@ SELECT logicalrelid, repmodel FROM pg_dist_partition WHERE logicalrelid = 'mx_te
+
+CREATE TABLE mx_table_with_small_sequence(a int, b SERIAL);
+SELECT create_distributed_table('mx_table_with_small_sequence', 'a');
+
 SELECT start_metadata_sync_to_node('localhost', :worker_1_port);
+DROP TABLE mx_table_with_small_sequence;
+SELECT start_metadata_sync_to_node('localhost', :worker_1_port);
+
 
+-- Create an MX table with (BIGSERIAL) sequences
 CREATE TABLE mx_table_with_sequence(a int, b BIGSERIAL, c BIGSERIAL);
 SELECT create_distributed_table('mx_table_with_sequence', 'a');
 \d mx_table_with_sequence
 \ds mx_table_with_sequence_b_seq
 \ds mx_table_with_sequence_c_seq
 
--- Check that the sequences created on the worker as well
+-- Check that the sequences created on the metadata worker as well
 \c - - - :worker_1_port
 \d mx_table_with_sequence
 \ds mx_table_with_sequence_b_seq
 \ds mx_table_with_sequence_c_seq
 
--- Check that dropping the mx table with sequences works as expected
+-- Check that the sequences on the worker have their own space
+SELECT nextval('mx_table_with_sequence_b_seq');
+SELECT nextval('mx_table_with_sequence_c_seq');
+
+-- Check that adding a new metadata node sets the sequence space correctly
 \c - - - :master_port
+SELECT start_metadata_sync_to_node('localhost', :worker_2_port);
+
+\c - - - :worker_2_port
+SELECT groupid FROM pg_dist_local_group;
+\d mx_table_with_sequence
+\ds mx_table_with_sequence_b_seq
+\ds mx_table_with_sequence_c_seq
+SELECT nextval('mx_table_with_sequence_b_seq');
+SELECT nextval('mx_table_with_sequence_c_seq');
+
+-- Check that dropping the mx table with sequences works as expected, even the metadata
+-- syncing is stopped to one of the workers
+\c - - - :master_port
+SELECT stop_metadata_sync_to_node('localhost', :worker_2_port);
 DROP TABLE mx_table_with_sequence;
 \d mx_table_with_sequence
 \ds mx_table_with_sequence_b_seq
 \ds mx_table_with_sequence_c_seq
 
--- Check that the sequences are dropped from the worker as well
+-- Check that the sequences are dropped from the workers
 \c - - - :worker_1_port
 \d mx_table_with_sequence
 \ds mx_table_with_sequence_b_seq
 \ds mx_table_with_sequence_c_seq
 
+-- Check that the sequences are dropped from the workers
+\c - - - :worker_2_port
+\ds mx_table_with_sequence_b_seq
+\ds mx_table_with_sequence_c_seq
+
+-- Check that MX sequences play well with non-super users
+\c - - - :master_port
+CREATE TABLE pg_dist_shard_placement_temp AS SELECT * FROM pg_dist_shard_placement;
+CREATE TABLE pg_dist_partition_temp AS SELECT * FROM pg_dist_partition;
+DELETE FROM pg_dist_shard_placement;
+DELETE FROM pg_dist_partition;
+SELECT master_remove_node('localhost', :worker_2_port);
+CREATE USER mx_user;
+\c - - - :worker_1_port
+CREATE USER mx_user;
+\c - - - :worker_2_port
+CREATE USER mx_user;
+
+\c - mx_user - :master_port
+CREATE TABLE mx_table (a int, b BIGSERIAL);
+SET citus.shard_replication_factor TO 1;
+SELECT create_distributed_table('mx_table', 'a');
+
+\c - postgres - :master_port
+SELECT master_add_node('localhost', :worker_2_port);
```

Comment:
Added comments there :) Basically we are testing that, if metadata is synced to a node which does not have the shards (therefore the sequence), the sequence creation (with a different user) would work as expected. For that, we need to remove the node from the list. 

We could do this with a `DELETE FROM pg_dist_node` command but the tests that rely on nodeid sequence should have set it explicitly anyway. Therefore, I have changed these :)